fix: security checks for team and user ids

NisanthChsr opened the pull request:
## Title

Fix the security check issue mentioned here https://github.com/BerriAI/litellm/issues/8394

<!-- e.g. "Implement user authentication feature" -->

## Relevant issues

Changed according to the comments

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the `tests/litellm/` directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on (`make test-unit`)[https://docs.litellm.ai/docs/extras/contributing_code]
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🐛 Bug Fix

## Changes
Ensured user_id is not reset during key updation calls

sairanga-ub committed:
fix: security checks for team and user ids

krrishdholakia commented:
> Tests are not working, need to check deeper to fix them



you're running tests in the wrong folder. Please add your tests inside `/tests/litellm` - https://github.com/BerriAI/litellm/tree/main/tests/litellm

krrishdholakia commented on litellm/proxy/management_endpoints/key_management_endpoints.py:
```diff
@@ -636,6 +638,31 @@ def prepare_key_update_data(
         non_default_values[k] = v
 
+    # Check if user is trying to modify user_id or team_id
+    if user_api_key_dict is not None and user_api_key_dict.user_role != LitellmUserRoles.PROXY_ADMIN.value:
```
can we refactor this into a separate function for easier testing. 

I believe there are already permission checks on key generate - is it possible to reuse code, and prevent duplication?

NisanthChsr replied:
Cleared out those checks, as they are present already.

When the existing key row contains a user_id, maintain the same value if the new data's user_id is Node (preventing it from resetting the value to None).

sairanga-ub committed:
fix: ensure user_id is not reset during key updation calls

sairanga-ub committed:
fix: nits

krrishdholakia commented:
This fix is confusing - so as a proxy admin, i would no longer be able to unset the user_id of a key? 

Why not just fix the UI to not pass a null value unless specified by user?

sairanga-ub committed:
fix: admin can unset the user_id of key

NisanthChsr commented:
> This fix is confusing - so as a proxy admin, i would no longer be able to unset the user_id of a key?
> 
> Why not just fix the UI to not pass a null value unless specified by user?

If I am not looking at the code in the right place, correct me

- I do not see in the UI where we can change the user_id during key update calls. 
- Since an admin should be able to unset the user_id, I modified the code accordingly in the backend.
- I added commented code for reference in the UI, where we can update the condition in the front-end. If needed, I'll make the modifications accordingly.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=9192) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=9192) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=9192) it.</sub>
